compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/declarations/IrDeclarationOrigin.kt

```diff
@@ -37,6 +37,7 @@ interface IrDeclarationOrigin {
     object MOVED_EXTENSION_RECEIVER : IrDeclarationOriginImpl("MOVED_EXTENSION_RECEIVER")
 
     object FILE_CLASS : IrDeclarationOriginImpl("FILE_CLASS")
+    object MULTIFILE_PART_CLASS : IrDeclarationOriginImpl("MULTIFILE_PART_CLASS", isSynthetic = true)
```

Comment:
We have another compilation mode of multifile class facades and parts, enabled by `-Xmultifile-parts-inherit` and used for example in kotlin-stdlib, where instead of generating delegating "bridges" in the facade, we just sort all the parts and inherit them from each other, and then inherit the facade from the last part.

In that mode I believe the parts should _not_ be synthetic, since otherwise methods declared there would not be visible from Java. I'm not sure why there are no tests on this (maybe because we use kotlin-stdlib itself as a test case).

Reply:
You're right, I didn't think about this at all. I've added the special case for `-Xmultifile-parts-inherit` and also added a test for this.

I've also renamed the corresponding origin to `SYNTHETIC_FILE_CLASS`, since not all multifile parts will use this origin now.